shared/src/metabase/mbql/schema.cljc

```diff
@@ -460,7 +466,8 @@
 
 (def arithmetic-expressions
   "Set of valid arithmetic expression clause keywords."
-  #{:+ :- :/ :* :coalesce :length :round :ceil :floor :abs :power :sqrt :log :exp :case})
+  #{:+ :- :/ :* :coalesce :length :round :ceil :floor :abs :power :sqrt :log :exp :case
+    :datediff})
```

Comment:
I'm not 100% sure of this. Is `arithmetic-expressions` things that operate on numbers, or things that _return_ numbers. I'm working on the latter. So `datediff` returns a number and is fine to use with any of the other arithmetic expressions.

Reply:
They are things that return numbers, so this is correct. See for example its use in NumericExpressionArg [here](https://github.com/metabase/metabase/blob/9141582508170bdbe99c4aa74bc4444e41475be6/shared/src/metabase/mbql/schema.cljc#L505).